pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedConsumerImpl.java

```diff
@@ -247,6 +268,8 @@ protected Message internalReceive(int timeout, TimeUnit unit) throws PulsarClien
                     if (completed.decrementAndGet() == 0) {
                         if (unsubscribeFail.get() == null) {
                             setState(State.Closed);
+                            unAckedMessageTracker.close();
```

Comment:
Maybe we could have the `close()` operation to imply a `clear()` as well.

Reply:
done